multi-agent/lib/mas/elements/sandboxes/openshell_sandbox/openshell_sandbox.py

```
@@ -0,0 +1,295 @@
+        """
+        for url in urls:
+            parsed = urlparse(url)
+            host = parsed.hostname or ""
+            port = parsed.port or (443 if parsed.scheme == "https" else 80)
+            if host:
+                self._allowed_endpoints.add((host, port))
+
+    _DEFAULT_ENDPOINTS: List[Tuple[str, int]] = [
+        ("pypi.org", 443),
+        ("files.pythonhosted.org", 443),
+    ]
```

Ruff (0.15.20)

[warning] Mutable default value for class attribute

(RUF012)